kernels/portable/cpu/op_glu.cpp

```diff
@@ -120,11 +34,63 @@ void slice_tensor(
  *  2. The output shall be in float types (Float, Double)
  */
 template <typename CTYPE_IN, typename CTYPE_OUT>
-Tensor& glu_out_tensor(const Tensor& self, int64_t dim, Tensor& out) {
+Tensor& glu_out_tensor(
+    KernelRuntimeContext& ctx,
+    const Tensor& self,
+    int64_t dim,
+    Tensor& out) {
   const auto self_size = self.size(dim);
-  slice_tensor<CTYPE_IN, CTYPE_OUT>(self, dim, self_size / 2, self_size, out);
-  sigmoid_tensor<CTYPE_OUT>(out);
-  mul_tensors<CTYPE_IN, CTYPE_OUT>(self, dim, out);
+  ET_KERNEL_CHECK(
+      ctx,
+      self.dim() <= static_cast<ssize_t>(kTensorDimensionLimit),
+      InvalidArgument,
+      out);
+  std::array<executorch::aten::SizesType, kTensorDimensionLimit> half_sizes;
+  std::copy(self.sizes().begin(), self.sizes().end(), half_sizes.begin());
+  half_sizes[dim] /= 2;
+  TensorImpl first_half_impl(
+      self.scalar_type(),
+      self.dim(),
+      half_sizes.data(),
+      self.mutable_data_ptr(),
+      const_cast<executorch::aten::DimOrderType*>(self.dim_order().data()),
+      const_cast<executorch::aten::StridesType*>(self.strides().data()),
+      self.shape_dynamism());
+  TensorImpl second_half_impl(
+      self.scalar_type(),
+      self.dim(),
+      half_sizes.data(),
+      reinterpret_cast<char*>(self.mutable_data_ptr()) +
+          self.strides()[dim] * self_size / 2 * self.element_size(),
+      const_cast<executorch::aten::DimOrderType*>(self.dim_order().data()),
+      const_cast<executorch::aten::StridesType*>(self.strides().data()),
+      self.shape_dynamism());
+  Tensor first_half(&first_half_impl);
+  Tensor second_half(&second_half_impl);
+  ScalarType compute_type =
+      executorch::runtime::isFloatingType(self.scalar_type())
+      ? self.scalar_type()
+      : ScalarType::Float;
+  // @lint-ignore CLANGTIDY facebook-hte-CArray
+  static constexpr const char op_name[] = "glu.out";
+  ET_SWITCH_FLOATHBF16_TYPES(compute_type, ctx, op_name, CTYPE_COMPUTE, [&]() {
+    utils::apply_bitensor_elementwise_fn<
+        CTYPE_COMPUTE,
+        op_name,
+        utils::SupportedTensorDtypes::FLOATHBF16>(
+        [](const auto val_a, const auto val_b) -> CTYPE_COMPUTE {
+          // TODO: rewrite this to be vectorization-capable.
+          const auto one = static_cast<decltype(val_a)>(1.0);
+          return val_a * (one / (one + std::exp(-val_b)));
+        },
+        ctx,
+        first_half,
+        utils::SupportedTensorDtypes::FLOATHBF16,
+        second_half,
+        utils::SupportedTensorDtypes::FLOATHBF16,
+        out,
+        utils::internal::SupportNoncontiguousTensors());
```

Comment:
why didn't you pass the support_non_contiguous_tensors here as a template parameter?

Reply:
the argument exists because specifying it as a template parameter is messy; see other thread